pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -214,7 +214,12 @@ nixBuild() {
 }
 
 nixFlakeBuild() {
-    if [ -z "$buildHost" ]; then
+    if [[ -z "$buildHost" && -z "$targetHost" ]] &&
```

Comment:
It seems that effect of this `if` is just where we create symlink. We still run `nix build` locally. And we always produce local symlink. And destination of that symlink being used by caller of this function, apparently.

Is there any reason to have different behavior for `nixos-rebuild build` depending on where build/target?  
Manual says that `build` will leave a symlink for user.  
What is expected usage of `nixos-rebuild build` with build/target specified?  
*(I would expect that user inspects `./result`, maybe run VM, and then decide to run `./result/bin/*` script for test/boot/switch with target specified)*

P.S. I'm new to NixOS and most likely miss a lot of context :slightly_smiling_face: